src/core/thread/mle_types.hpp

```diff
@@ -428,6 +430,61 @@ class DeviceMode : public Equatable<DeviceMode>
     uint8_t mMode;
 };
 
+#if OPENTHREAD_FTD
+/**
+ * This class represents device configurations.
+ *
+ * The device configurations are used for calculating the local leader weight on the device.
+ *
+ */
+class DeviceConfig : public otDeviceConfig, public Clearable<DeviceConfig>
+{
+public:
+    /**
+     * This enumeration represents the device's power supply config.
+     *
+     */
+    enum PowerSupply : uint8_t
+    {
+        kPowerSupplyBattery          = OT_POWER_SUPPLY_BATTERY,           ///< Battery powered.
+        kPowerSupplyExternal         = OT_POWER_SUPPLY_EXTERNAL,          ///< External powered.
+        kPowerSupplyExternalStable   = OT_POWER_SUPPLY_EXTERNAL_STABLE,   ///< Stable external power with backup.
+        kPowerSupplyExternalUnstable = OT_POWER_SUPPLY_EXTERNAL_UNSTABLE, //< Unstable external power.
+    };
+
+    /**
+     * This constructor initializes `DeviceConfig` with default values.
+     *
+     */
+    DeviceConfig(void);
+
+    /**
+     * This method calculates the leader weight based on the device config.
+     *
+     * @returns The calculated leader weight.
+     *
+     */
+    uint8_t CalculateLeaderWeight(void) const;
+
+private:
+    static constexpr int8_t  kDefaultAdjustment        = OPENTHREAD_CONFIG_MLE_DEFAULT_LEADER_WEIGHT_ADJUSTMENT;
+    static constexpr uint8_t kBaseWeight               = 64;
+    static constexpr int8_t  kBorderRouterInc          = +1;
+    static constexpr int8_t  kCcmBorderRouterInc       = +8;
+    static constexpr int8_t  kIsUnstableInc            = -4;
+    static constexpr int8_t  kPowerBatteryInc          = -8;
+    static constexpr int8_t  kPowerExtenralInc         = 0;
```

Comment:
```suggestion
    static constexpr int8_t  kPowerExternalInc         = 0;
```

Reply:
Change also the typo in other 'extenral' items.

Reply:
Thanks. Fixed them in new push